kops/libsonnet/cluster.jsonnet

```diff
@@ -1,4 +1,33 @@
-// local cluster(name, configBase, zone, masterIgName, networkCIDR, subnets) = {
+// Exports a customizable kops Cluster object.
+// https://kops.sigs.k8s.io/cluster_spec/ lists available properties.
+//
+// The default configuration sets up the following:
+//
+// 1. One etcd cluster each on the master node for events & api,
+//    with minimal resource allocations
+// 2. Calico for in-cluster networking https://kops.sigs.k8s.io/networking/calico/,
+//    with the default settings. Explicitly decided against AWS-VPC cluster networking
+//    due to pod density issues - see https://github.com/2i2c-org/pangeo-hubs/issues/28.
+// 3. Nodes in only one subnet in one AZ. Ideally, the master would be multi-AZ but
+//    the nodes would be single AZ. Multi AZ workers run into problems attaching PVs
+//    from other AZs (for hub db PVC, for example), and incurs networking cost for no
+//    clear benefit in our use case. An opinionated set of IP ranges is picked here,
+//    and the subnet is created in _config.zone.
+// 4. A /16 network for the entire cluster, with a /19 allocated to the one subnet
+//    currently in use. This allows for ~8000 currently active pods.
+//    FIXME: Consider a /18 instead?
```

Comment:
Can you elaborate a little bit more about these decisions? 
Why a /16 network for the entire cluster?
Btw, /19 (~ 8000 pods seems) seems a sensible number for now, IMHO.

Reply:
@damianavila ah, these are just the kops defaults! I'll document that too.

Reply:
Updated!

Reply:
Since we are on this discussion :wink: what is the network story in GKE?
Looking at this one right now: https://cloud.google.com/kubernetes-engine/docs/how-to/flexible-pod-cidr
Are we using the default values?

Reply:
Yeah, we're just using the default values. This is from terraform in this repo at `terraform/`

Reply:
Yep, I just wanted to double-check because I looked into the terraform stuff in this repo and I did not find any customization (and sometimes things are deployed on the fly and we - at least I - forgot about them 😉 ).